lib/spack/spack/compiler.py

```diff
@@ -251,31 +250,28 @@ def check(exe):
     def version(self):
         return self.spec.version
 
-    @classmethod
-    def verbose_flag(cls):
-        """
-        This property should be overridden in the compiler subclass if a
-        verbose flag is available.
-
-        If it is not overridden, it is assumed to not be supported.
-        """
-
-    @classmethod
-    def parse_implicit_rpaths(cls, string):
-        """Parses link paths out of compiler debug output.
+    def implicit_rpaths(self):
+        if self.enable_implicit_rpaths is False:
+            return []
 
-        Args:
-            string (str): compiler debug output as a string
+        exe_paths = [
+            x for x in [self.cc, self.cxx, self.fc, self.f77] if x]
+        link_dirs = self._get_compiler_link_paths(exe_paths)
 
-        Returns:
-            (list of str): implicit link paths parsed from the compiler output
+        all_required_libs = (
+            list(self.required_libs) + Compiler._all_compiler_rpath_libraries)
+        return list(paths_containing_libs(link_dirs, all_required_libs))
 
-        Subclasses can override this to customize.
+    @property
+    def required_libs(self):
+        """For executables created with this compiler, the compiler libraries
+        that would be generally required to run it.
         """
-        return _parse_implicit_rpaths(string)
+        # By default every compiler returns the empty list
+        return []
 
     @classmethod
-    def determine_implicit_rpaths(cls, paths):
+    def _get_compiler_link_paths(cls, paths):
```

Comment:
This method should move outside of the class and be a free method on the module, since it doesn't use the `cls` variable anywhere (the one place it does is to call a method I've also asked you to move outside the class).